### ISSUE TITLE
Update failed with hasManyThrough relationship

### ISSUE THREAD

**Issue by AminulBD** [laravel/framework#25740]
- Laravel Version: v5.6.38
- PHP Version: 7.2.8
- Database Driver & Version: MySQL v5.7.23

### Description:
I've made a relationship with hasManyThrough and it's working perfectly to fetch data. But, When I'm trying to update I get this error below.

````
Integrity constraint violation: 1052 Column 'customer_id' in field list is ambiguous (SQL: update `invoice_items` inner join `invoices` on `invoices`.`id` = `invoice_items`.`invoice_id` set `customer_id` = 2, `user_id` = 7, `updated_at` = 2018-09-22 21:47:15 where `invoices`.`order_id` = 17)
````

### Steps To Reproduce:
**Tables structures:**
````
orders
- id
- user_id
- customer_id
- content
invoices
- id
- order_id
- user_id
- customer_id
- content
transactions
- id
- invoice_id
- user_id
- customer_id
- content
````

**Relationships**
````
public function transactions()
{
   return $this->hasManyThrough('App\Transaction', 'App\Invoice');
}
````
**What i expected to work**
````
$order->transactions()->update([
    'customer_id' => 100,
    'user_id' => 100
]);
````

**Comment by AminulBD** [comment]
I've found a temporary solution about this issue like this: 

````
foreach ($order->transactions as $transaction) {
    $transaction->update($updateRules);
}
````

**Comment by staudenmeir** [comment]
Specify the table name:

```php
$order->transactions()->update([
    'transactions.customer_id' => 100,
    'transactions.user_id' => 100
]);
```

**Comment by AminulBD** [comment]
@standaniels Thank you. But, you know if I specified the table name but the problem still exists because the `updated_at` which can't be set by specific table name.

Have a look at this error:
````
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'updated_at' in field list is ambiguous (SQL: update `invoice_items` inner join `invoices` on `invoices`.`id` = `invoice_items`.`invoice_id` set `invoice_items`.`customer_id` = 2, `invoice_items`.`user_id` = 3, `invoice_items`.`updated_at` = 2018-09-23 21:33:54, `updated_at` = 2018-09-23 21:33:54 where `invoices`.`order_id` = 25)
````

I've experimented with below code by adding `transactions.updated_at => now()->toDateTimeString()`. Still no luck.

**Comment by staudenmeir** [comment]
This is a more complex issue, see #13909.

**Comment by staudenmeir** [comment]
You can use [this](https://github.com/laravel/framework/issues/13909#issuecomment-367657638) workaround:

```php
$order->transactions()->toBase()->update([
    'transactions.customer_id' => 100,
    'transactions.user_id' => 100,
    'transactions.updated_at' => now()
]);
```

**Comment by staudenmeir** [comment]
Please close the issue.

**Comment by staudenmeir** [comment]
Will be fixed in Laravel 5.8: #26031
